src/state/opamSwitchState.ml

```diff
@@ -1005,7 +1005,7 @@ let universe st
       OpamPackage.Set.empty
   in
   let avoid_versions =
-    OpamPackage.Set.filter (avoid_version st) u_available
+    OpamPackage.Set.filter (avoid_version st) st.packages
```

Comment:
This set could then be intersected with `u_available` in `OpamSolver.opam2cudf_map` which would then make the usage of the set _exactly_ the same (`u_available` is also forced in `OpamSolver.opam2cudf_map`, so the only cost is the set intersection)

Reply:
done